deps/spidershim/spidermonkey/mozglue/build/moz.build

```diff
@@ -6,10 +6,7 @@
 
 # Build mozglue as a shared lib on Windows, OSX and Android.
 # If this is ever changed, update MOZ_SHARED_MOZGLUE in browser/installer/Makefile.in
-if CONFIG['OS_TARGET'] in ('WINNT', 'Darwin', 'Android'):
-    SharedLibrary('mozglue')
-else:
-    Library('mozglue')
+SharedLibrary('mozglue')
```

Comment:
The build expects libmozglue.so to be in deps/spidershim/build/dist/bin, but on Linux because we have Library('mozglue') we produce libmozglue.a.desc instead.

This is why the build was working on your Mac but not for people building on Linux.